src/javascript/expression.rs

```diff
@@ -172,6 +187,9 @@ pub fn constant_expression<'a, T, Y>(expression: &'a Constant<T, Y>) -> Output<'
         Constant::String { value, .. } => Ok(string(&value.as_str())),
         Constant::Tuple { elements, .. } => array(elements.iter().map(|e| constant_expression(&e))),
         Constant::List { .. } => unsupported("List as constant"),
+        Constant::Record { name, .. } if name == "True" => Ok("true".to_doc()),
```

Comment:
Is checking just the name enough? I was looking that the module field in this `Constant::Record` struct and it was marked as None so I couldn't see anything that would flag this as Prelude.

Is it even possible to define a Custom type with varients True/False/Nil?

Reply:
It is possible to use those names with custom types, so we need to check that the type, which is stored on the record's `typ` field.

`Type` has methods like `is_int`, we can add an `is_bool` and `is_nil` to match.

Reply:
How come in the ast module there is a single constant module. which has typed untyped variants. https://github.com/gleam-lang/gleam/blob/main/src/ast/constant.rs#L4-L5

Whereas for expressions there are separate data structures for the typed, untyped variants?

Reply:
Also it looks like the details of an `is_false` function does simply check the name and that the module defining it is none. https://github.com/gleam-lang/gleam/blob/main/src/type_.rs#L93

Still neater to put it on the Type so will do that.

Reply:
> How come in the ast module there is a single constant module. which has typed untyped variants. https://github.com/gleam-lang/gleam/blob/main/src/ast/constant.rs#L4-L5
> Whereas for expressions there are separate data structures for the typed, untyped variants?

This is because the typed and untyped AST have a different structure, as more has been learnt about the code during type analysis. This isn't the case for constants (at least not to an extent we can't represent it with generics) so it has one parameterised AST.

Reply:
Thanks for the explanation. I think PR is done apart from null vs undefined.